edge-hub/test/Microsoft.Azure.Devices.Edge.Hub.Core.Test/ConnectionManagerTest.cs

```diff
@@ -733,6 +761,8 @@ public async Task KeepSubscriptionsOnDeviceRemoveTest()
             Assert.Equal(2, subscriptions.Count);
             Assert.True(subscriptions[DeviceSubscription.Methods]);
             Assert.True(subscriptions[DeviceSubscription.C2D]);
+            Option<ICloudProxy> cloudProxy = await connectionManager.GetCloudConnection(deviceId);
+            Assert.True(cloudProxy.HasValue);
```

Comment:
I would love to also prove that after the initial AddDeviceConnection on line 727, we DON'T have a CloudConnection for that device. I'm not so sure how to do that, since GetCloudConnection will create one.